File: app/src/main/java/com/res/jobjob/MainActivity.java
package com.res.jobjob;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.google.firebase.auth.FirebaseAuth;
import com.res.jobjob.activities.Clientes.MapCliente;
import com.res.jobjob.activities.Login;
import com.res.jobjob.activities.OptAuth;
import com.res.jobjob.activities.Socios.MapSocio;

public class MainActivity extends AppCompatActivity {

    Button btnSoySocio, btnSoyCliente;
    SharedPreferences mPref;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnSoyCliente = findViewById(R.id.btnSoyCliente);
        btnSoySocio = findViewById(R.id.btnSoySocio);

        mPref = getApplicationContext().getSharedPreferences("typeUser",MODE_PRIVATE);
        final SharedPreferences.Editor editor= mPref.edit();


        btnSoySocio.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editor.putString("user","socio");
                editor.apply();
                Intent i = new Intent(MainActivity.this, OptAuth.class);
                startActivity(i);

            }
        });

        btnSoyCliente.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                editor.putString("user","cliente");
                editor.apply();
                startActivity( new Intent(MainActivity.this,OptAuth.class));
            }
        });

    }

    @Override
    protected void onStart() {
        super.onStart();
        if(FirebaseAuth.getInstance().getCurrentUser() != null){
            String user = mPref.getString("user","");
            if(user.equals("cliente")){

                Intent intent = new Intent(MainActivity.this, MapCliente.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                startActivity(intent);

            }else{
                Intent intent = new Intent(MainActivity.this, MapSocio.class);
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                startActivity(intent);


            }
        }
    }
}

File: app/src/main/java/com/res/jobjob/MainActivity.kt
package com.res.jobjob

import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.auth.FirebaseAuth
import com.res.jobjob.activities.clientes.MapCliente
import com.res.jobjob.activities.OptAuth
import com.res.jobjob.activities.socios.MapSocio

class MainActivity : AppCompatActivity() {

    private lateinit var btnSoySocio: Button
    private lateinit var btnSoyCliente: Button
    private lateinit var mPref: SharedPreferences

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btnSoyCliente = findViewById(R.id.btnSoyCliente)
        btnSoySocio = findViewById(R.id.btnSoySocio)
        mPref = applicationContext.getSharedPreferences("typeUser", MODE_PRIVATE)
        val editor = mPref.edit()
        btnSoySocio.setOnClickListener {
            editor.putString("user", "socio")
            editor.apply()
            val i = Intent(this@MainActivity, OptAuth::class.java)
            startActivity(i)
        }
        btnSoyCliente.setOnClickListener {
            editor.putString("user", "cliente")
            editor.apply()
            startActivity(Intent(this@MainActivity, OptAuth::class.java))
        }
    }

    override fun onStart() {
        super.onStart()
        if (FirebaseAuth.getInstance().currentUser != null) {
            val user = mPref.getString("user", "")
            if (user == "cliente") {
                val intent = Intent(this@MainActivity, MapCliente::class.java)
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
                startActivity(intent)
            } else {
                val intent = Intent(this@MainActivity, MapSocio::class.java)
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
                startActivity(intent)
            }
        }
    }
}